javatests/arcs/android/systemhealth/testapp/TestStorageService.kt

```diff
@@ -0,0 +1,69 @@
+package arcs.android.systemhealth.testapp
+
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import arcs.android.storage.ParcelableStoreOptions
+import arcs.sdk.android.storage.service.StorageService
+import arcs.sdk.android.storage.service.StorageServiceBindingDelegate
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlin.random.Random
+
+/**
+ * Arcs system-health-test storage service. Supports crashing itself when needed.
+ */
+class TestStorageService : StorageService() {
+    private val coroutineContext = Dispatchers.Main + CoroutineName("TestStorageService")
+    private val scope = CoroutineScope(coroutineContext)
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        if (intent != null &&
+            intent.hasExtra(EXTRA_CRASH) &&
+            intent.getBooleanExtra (EXTRA_CRASH, false)) {
+            scope.launch {
+                delay(Random.nextLong(10, 1000))
+                android.os.Process.killProcess(android.os.Process.myPid());
+            }
+        }
+        return super.onStartCommand(intent, flags, startId)
+    }
+
```

Comment:
Could you also add one liner:
override fun onDestroy() = scope.cancel()

Reply:
Done.

Reply:
Done.